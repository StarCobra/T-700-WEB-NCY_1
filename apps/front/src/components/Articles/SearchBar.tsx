import React, { useState } from "react";
import "../../style/article.scss";
import { Box } from "@mui/material";
import Select from "../Select";

export default function SearchBar() {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [keywords, setKeywords] = useState([]);

  // TODO : remplacer par valeur de l'API
  const options = [
    { value: "Income", label: "Income" },
    { value: "BTC", label: "BTC" },
    { value: "Bankrupt", label: "Bankrupt" },
    { value: "ETH", label: "ETH" },
  ];

  return (
    <form className="searchBarContainer">
      <Box className="selectSearchContainer">
        <Box className="selectSearch">
          <Select
            name="Select"
            label="Choose a keyword"
            options={options}
            handleChange={(e: any) => setKeywords(e.target.value)}
          />
        </Box>
      </Box>
    </form>
  );
}
