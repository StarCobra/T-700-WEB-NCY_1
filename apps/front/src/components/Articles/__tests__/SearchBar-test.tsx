import React from "react";
import { act, create } from "react-test-renderer";
import SearchBar from "../SearchBar";

describe("Components > Articles > SearchBar", () => {
  const tests = [
    {
      description: "without props",
      props: {},
    }
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(<SearchBar {...test.props} />);
      });
      expect(root).toMatchSnapshot();
    });
  });
});
