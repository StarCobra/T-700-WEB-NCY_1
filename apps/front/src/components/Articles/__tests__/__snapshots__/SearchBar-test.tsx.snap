// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Articles > SearchBar should render the component generic 1`] = `
<form
  className="searchBarContainer"
>
  <div
    className="selectSearchContainer MuiBox-root css-0"
  >
    <div
      className="selectSearch MuiBox-root css-0"
    >
      <div
        className="selectCryptoContainer MuiBox-root css-0"
      >
        <p>
          Choose a keyword
        </p>
        <div
          className="SelectCryptoBox MuiBox-root css-4stev7"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root SelectCryptoLabel css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorWarning MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mk9tez-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r0:"
              id=":r0:-label"
            >
              Choose a keyword
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorWarning MuiInputBase-fullWidth MuiInputBase-formControl css-nmbvik-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <div
                aria-controls=":r1:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r0:-label :r0:"
                className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="combobox"
                tabIndex={0}
              >
                <span
                  className="notranslate"
                >
                  â€‹
                </span>
              </div>
              <input
                aria-hidden={true}
                aria-invalid={false}
                autoFocus={false}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                name="Select"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Choose a keyword
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
